atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -271,38 +271,39 @@ private static void checkAndSetReplicationFactor(
                     + " and running the appropriate repairs."
                     + " Contact the AtlasDB team to perform these steps.";
             if (currentRf != 1) {
-                logErrorOrThrow(errorMessage, safetyDisabled);
+                logErrorOrThrow(errorMessage, config.safetyDisabled() || config.ignoreBadNodeTopologyChecks());
             }
             // Automatically convert RF=1 to look like network partitioner.
-            dcs = sanityCheckDatacenters(client, desiredRf, safetyDisabled);
+            dcs = sanityCheckDatacenters(client, config);
             if (dcs.size() > 1) {
-                logErrorOrThrow(errorMessage, safetyDisabled);
+                logErrorOrThrow(errorMessage,
+                        config.safetyDisabled() || config.ignoreBadDatacenterConfigurationChecks());
             }
-            if (!safetyDisabled) {
+            if (!(config.safetyDisabled() || config.ignoreBadDatacenterConfigurationChecks())) {
                 ks.setStrategy_class(CassandraConstants.NETWORK_STRATEGY);
                 ks.setStrategy_options(ImmutableMap.of(dcs.iterator().next(), "1"));
             }
         } else {
-            dcs = sanityCheckDatacenters(client, desiredRf, safetyDisabled);
+            dcs = sanityCheckDatacenters(client, config);
         }
 
-        sanityCheckReplicationFactor(ks, desiredRf, safetyDisabled, dcs);
+        sanityCheckReplicationFactor(ks, config, dcs);
     }
 
-    static void currentRfOnKeyspaceMatchesDesiredRf(Client client, CassandraKeyValueServiceConfig config,
-            boolean safetyDisabled) throws TException {
+    static void currentRfOnKeyspaceMatchesDesiredRf(Client client, CassandraKeyValueServiceConfig config)
+            throws TException {
         KsDef ks = client.describe_keyspace(config.keyspace());
-        Set<String> dcs = sanityCheckDatacenters(client, config.replicationFactor(), safetyDisabled);
-        sanityCheckReplicationFactor(ks, config.replicationFactor(), safetyDisabled, dcs);
+        Set<String> dcs = sanityCheckDatacenters(client, config);
+        sanityCheckReplicationFactor(ks, config, dcs);
     }
 
-    private static void sanityCheckReplicationFactor(KsDef ks, int desiredRf, boolean safetyDisabled, Set<String> dcs) {
+    private static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
         Map<String, String> strategyOptions = Maps.newHashMap(ks.getStrategy_options());
         for (String dc : dcs) {
             if (strategyOptions.get(dc) == null) {
                 logErrorOrThrow("The datacenter for this cassandra cluster is invalid. "
                         + " failed dc: " + dc
-                        + "  strategyOptions: " + strategyOptions, safetyDisabled);
+                        + "  strategyOptions: " + strategyOptions, config.ignoreBadDatacenterConfigurationChecks());
```

Comment:
Should this be `config.safetyDisabled() || config.ignoreBadDatacenterConfigurationChecks()`?